cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -115,6 +115,43 @@ public void testAppWithConfig() throws Exception {
     deleteArtifact(artifactId, 200);
   }
 
+  @Test
+  public void testAppWithConfigurationString() throws Exception {
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "ConfigApp");
+    Id.Artifact artifactId = Id.Artifact.from(Id.Namespace.DEFAULT, "appWithConfig", "1.0.0-SNAPSHOT");
+    HttpResponse response = addAppArtifact(artifactId, ConfigTestApp.class);
+    Assert.assertEquals(200, response.getResponseCode());
+
+    ConfigTestApp.ConfigClass configuration = new ConfigTestApp.ConfigClass("abc", "def");
+    String configurationString = GSON.toJson(configuration);
+    response = deploy(appId, new AppRequest<>(ArtifactSummary.from(artifactId.toArtifactId()), null, null, null, null,
+                                              configurationString));
```

Comment:
just want to check - this test will fail without the fix right?

Reply:
yes I removed the change and ran this test and it failed